dash/App/Accounts/Add/AddRing/index.js

```diff
@@ -1,208 +1,29 @@
-            >
-              Add Keyring Account
-            </div>
-            <div
-              className='addAccountItemOptionSetup'
-              style={{ transform: `translateX(-${100 * this.state.index}%)` }}
-            >
-              <div className='addAccountItemOptionSetupFrames'>
-                <div className='addAccountItemOptionSetupFrame'>
-                  <div className='addAccountItemOptionTitle'>Enter Private Key</div>
-                  <div className='addAccountItemOptionInputPhrase'>
-                    <input
-                      type='password'
-                      tabIndex='-1'
-                      ref={this.forms.enterPrivateKey}
-                      value={this.state.privateKey}
-                      onChange={(e) => this.onChange('privateKey', e)}
-                      onFocus={(e) => this.onFocus('privateKey', e)}
-                      onBlur={(e) => this.onBlur('privateKey', e)}
-                      onKeyPress={(e) => this.keyPress(e, () => this.next())}
-                    />
-                  </div>
-                  <div className='addAccountItemOptionSubmit' onClick={() => this.next()}>
-                    Next
-                  </div>
-                </div>
-                <div className='addAccountItemOptionSetupFrame'>
-                  <div className='addAccountItemOptionTitle'>Create Password</div>
-                  <div className='addAccountItemOptionInputPhrase addAccountItemOptionInputPassword'>
-                    <div className='addAccountItemOptionSubtitle'>
-                      password must be 12 characters or longer
-                    </div>
-                    <input
-                      type='password'
-                      tabIndex='-1'
-                      ref={this.forms.manualCreatePassword}
-                      value={this.state.password}
-                      onChange={(e) => this.onChange('password', e)}
-                      onFocus={(e) => this.onFocus('password', e)}
-                      onBlur={(e) => this.onBlur('password', e)}
-                      onKeyPress={(e) => this.keyPress(e, () => this.createManual())}
-                    />
-                  </div>
-                  <div className='addAccountItemOptionSubmit' onClick={() => this.createManual()}>
-                    Create
-                  </div>
-                </div>
-                <div className='addAccountItemOptionSetupFrame'>
-                  <div className='addAccountItemOptionTitle'>{this.state.status}</div>
-                  {this.state.error ? (
-                    <div className='addAccountItemOptionSubmit' onClick={() => this.restart()}>
-                      try again
-                    </div>
-                  ) : null}
-                </div>
-              </div>
-            </div>
-          </div>
-          <div className='addAccountItemFooter' />
-        </div>
-      </div>
-    )
-  }
+const isValidSecret = (privateKeyStr) => {
```

Comment:
love this abstraction of common code. we don't need to unit test everything that wasn't unit tested before but can we add simple unit tests for these little bits of logic to make sure we're covering common error cases and our code isn't throwing any errors? 
`it should handle an invalid hex string`, `it should handle an invalid private key`, etc

lmk if you need any help setting this up